connection.go

```diff
@@ -40,9 +59,30 @@ func (ws *Ws) connect() (err error) {
 		ws.host = ws.host + "/gremlin"
 		ws.conn, _, err = d.Dial(ws.host, http.Header{})
 	}
+
+	if err == nil {
+		ws.connected = true
+		ws.conn.SetPongHandler(func(appData string) error {
+			ws.Lock()
+			ws.connected = true
+			ws.Unlock()
+			return nil
+		})
+	}
 	return
 }
 
+func (ws *Ws) isConnected() bool {
+	defer ws.Unlock()
```

Comment:
You should be able to get rid of the lock thus reducing overhead. Since the function only returns a value calling lock and relying on `Unlock` to be called from the stack before the function returns does nothing.